### ISSUE TITLE
Going crazy with --watch

### ISSUE THREAD

**Issue by sir-marc** [adamgruber/mochawesome#76]
If i add `--watch` flag, the reporter is going crazy.
It looks like he is doing all tests multiple times and adds and removes the css several times.
Sometimes they are missing at the end of the crazy-going-session.

**Comment by adamgruber** [comment]
Hmm, thanks for the find. I don't think I ever tested this with the `--watch` flag on. I'll have to look into it more.

**Comment by sir-marc** [comment]
made some further investigations:
The error occurs every time when i execute the mocha command with the --watch flag and there are alle css and font files. If i delete them before executing the command, it works. So every second call fails.
I'm just about to try if it has something to do with the custom output folder.

**Comment by zhouhua** [comment]
I have the same problem. And cleaning output folder works for me.

I find if there're some test case "slow", the bug occurs.

**Comment by adamgruber** [comment]
Is this still happening with v2? Can you provide some test code?

**Comment by zhouhua** [comment]
Sure.

spec file: 
```js
describe('slow cases', function () {
    it('is really slow', function (done) {
        setTimeout(done, 1000);
    });
});
```

script: 
```sh
mocha -w -R mochawesome test.spec.js
```

environment:
```
Node.js: v7.2.0
mocha: v3.2.0
mochawesome: v2.0.2
```

This bug occurs when some tests taking a long time (> 900ms). If I reduce the duration to 500ms, it would just test serval times again, not keep redoing it all the time.

**Comment by adamgruber** [comment]
Did some research on this. The issue comes from how mocha determines what files to watch. Mocha will watch all `.js` files inside `cwd`. If it's the first time running tests with mochawesome you're fine but every time after mocha will find the `app.js` file created by mochawesome and add it to the list of watched files. Since that file gets created each time the reporter runs, mocha picks it up as changed and runs the test again putting the whole process into an endless loop. Unfortunately mocha does not currently offer options to watch or ignore certain files.

As a workaround, you can run mochawesome with the `inlineAssets` option set to `true`. This prevents the `app.js` file from being created since all assets get inlined into the report HTML file.

I am going to look into changing mochawesome so it won't recreate the assets folder if it already exists.

**Comment by adamgruber** [comment]
This is fixed with mochawesome-report-generator@1.0.8. Assets only get copied once and no longer trigger infinite test re-runs.
